src/AutorestSwift/Code Generator/Logger.swift

```diff
@@ -43,32 +112,70 @@ class Logger {
 
     // MARK: Methods
 
-    func log(
-        _ message: @autoclosure @escaping () -> String?,
-        category: String = "default",
-        level: OSLogType = .default
-    ) {
+    func log(_ message: @autoclosure @escaping () -> String?, category: String = "default", level: LogLevel = .info) {
         guard let msg = message() else {
-            os_log("Unable to issue log message")
-            return
+            fatalError("Unable to create log message.")
         }
         if let logger = loggers[category] {
-            os_log("%@", log: logger, type: level, msg)
+            os_log("%@", log: logger, type: level.asOSLogLevel, msg)
         } else {
             let logger = OSLog(subsystem: name, category: category)
             loggers[category] = logger
-            os_log("%@", log: logger, type: level, msg)
+            os_log("%@", log: logger, type: level.asOSLogLevel, msg)
+        }
+
+    }
+
+    func logFailure(_ message: @autoclosure @escaping () -> String?, category: String = "default") -> Never {
+        guard let msg = message() else {
+            fatalError("Unable to create log message.")
         }
+        log(msg, category: category, level: .error)
+        fatalError(msg)
     }
 }
 
-class FileLogger {
+/// Cross-platform Swift logger implementation
+class SwiftLogger: Logger {
+
+    func log(_ message: @autoclosure @escaping () -> String?, category: String = "default", level: LogLevel = .info) {
+        guard let msg = message() else {
+            fatalError("Unable to create log message.")
+        }
+        // TODO: Implement
+    }
+
+    func logFailure(_ message: @autoclosure @escaping () -> String?, category: String = "default") -> Never {
+        guard let msg = message() else {
+            fatalError("Unable to create log message.")
+        }
+        // TODO: Implement
+        fatalError(msg)
+    }
+}
+
+/// Logs to a file.
+class FileLogger: Logger {
```

Comment:
Format looks like:
```
default (INFO) AutorestSwift.ChannelClient initializing
default (INFO) AutorestSwift.ChannelServer initializing
default (INFO) AutorestSwift.ChannelServer starting
default (INFO) Server Handler channelActive
default (INFO) --> decoding {"jsonrpc":"2.0","id":0,"method":"GetPluginNames","params":null}
default (INFO) Server Handler channelRead
default (INFO) rpc handler returned success list([AutorestSwift.RPCObject.string("swift")])
default (INFO) <-- encoding {"jsonrpc":"2.0","id":0,"result":["swift"]}
default (INFO) --> decoding 
```

I think it might make sense to replace "default" with "AutoRest.Swift".